### PR TITLE
[SourceKit] Only report textual results inside comments and strings from syntactic rename ranges

### PR DESCRIPTION
**Pull request by ahoppen**

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

File: lib/Refactoring/SyntacticRenameRangeDetails.cpp [hunk 1]
```diff
@@ -406,19 +406,18 @@ RegionType RenameRangeDetailCollector::addSyntacticRenameRanges(
...
-    // results (eg. from comments or string literals).
+    // results to cover comments or string literals.
     //
-    // Determine whether we should include them.
-    if (regionKind == RegionType::ActiveCode) {
-      // If the reference is in active code, we should have had a name usage
-      // from the index. Since we don't, they are likely unrelated symbols that
-      // happen to have the same name. Don't return them as matching ranges.
+    // We only want to include these textual matches inside comments and string
+    // literals. All other matches inside are likely bugus results.
```
Comment (bnbarham) [inline lib/Refactoring/SyntacticRenameRangeDetails.cpp:412]:
```suggestion
    // literals. All other matches inside are likely bogus results.
```

Though I had a good laugh at bugus, I kind of like it.

**Review by bnbarham — APPROVED** [review]

**Comment by bnbarham** [comment]
@swift-ci please build toolchain macOS platform

**Review by hamishknight — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please build toolchain macOS

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Pull request merged**